src/stream/src/task/barrier_manager/managed_state.rs

```diff
@@ -222,14 +260,19 @@ impl ManagedBarrierState {
                     },
                 ..
             }) => {
-                let remaining_actors = actor_ids_to_collect
+                let remaining_actors: HashSet<ActorId> = actor_ids_to_collect
                     .into_iter()
                     .filter(|a| !collected_actors.remove(a))
                     .collect();
+                for exited_actor in &self.exited_actors {
+                    if remaining_actors.contains(exited_actor) {
+                        bail!("Actor {} exit unexpectedly", exited_actor);
```

Comment:
We report the failure through the channel if an error occurs on `Issued`, while through the return value of this function if on `Stashed`. 🤔 Can we make the behavior more consistent?

Reply:
Unlikely, I'm afraid. 😢  If we do the check at the start of `send barrier` and `collect barrier`, we still need to report the failure through the channel at the time of collection, as actor exit may also occur while waiting for the collection to complete.